cdap-ui/app/cdap/components/FieldLevelLineage/v2/FllTable/index.tsx

```diff
@@ -90,74 +92,116 @@ const styles = (theme) => {
 
 interface ITableProps extends WithStyles<typeof styles> {
   tableId?: string;
-  fields?: IField[];
+  tableInfo?: ITableInfo;
   type?: string;
   isActive?: boolean;
 }
 
-function renderGridHeader(fields: IField[], isTarget: boolean, classes) {
+function renderGridHeader(
+  fields: IField[],
+  isTarget: boolean,
+  isExpanded: boolean = false,
+  classes
+) {
   const count: number = fields.length;
   const tableName = fields[0].dataset;
+  const options = { context: count };
   return (
     <div className={classes.tableHeader}>
       <div className="table-name">{tableName}</div>
       <div className={classes.tableSubheader}>
-        {isTarget
-          ? T.translate('features.FieldLevelLineage.v2.FllTable.fieldsCount', {
-              context: count,
-            })
-          : T.translate('features.FieldLevelLineage.v2.FllTable.relatedFieldsCount', {
-              context: count,
-            })}
+        {isTarget || isExpanded
+          ? T.translate('features.FieldLevelLineage.v2.FllTable.fieldsCount', options)
+          : T.translate('features.FieldLevelLineage.v2.FllTable.relatedFieldsCount', options)}
       </div>
     </div>
   );
 }
 
-function renderGridBody(fields: IField[], tableName: string, activeFields = new Set(), classes) {
+function renderGridBody(
+  fields: IField[],
+  tableId: string,
+  activeFields = new Set(),
+  hasUnrelatedFields: boolean = false,
+  isExpanded: boolean = false,
+  handleClick: () => void,
+  classes
+) {
+  const namespace = fields[0].namespace;
   return (
     <div
       className={classes.gridBody}
-      id={tableName}
-      data-tablename={fields[0].dataset}
-      data-namespace={fields[0].namespace}
+      id={tableId}
+      data-tablename={tableId}
+      data-namespace={namespace}
     >
       {fields.map((field) => {
         const isActiveField = activeFields.has(field.id);
         return <FllField key={field.id} field={field} isActive={isActiveField} />;
       })}
+      <If condition={hasUnrelatedFields}>
+        <ExpandableField
+          isExpanded={isExpanded}
+          handleClick={handleClick}
+          tablename={fields[0].dataset}
+        />
+      </If>
     </div>
   );
 }
 
-function FllTable({ tableId, fields, type, isActive, classes }: ITableProps) {
+function FllTable({ tableId, tableInfo, type, isActive, classes }: ITableProps) {
   const GRID_HEADERS = [{ property: 'name', label: tableId }];
-  const { target, activeCauseSets, activeImpactSets } = useContext<IContextState>(FllContext);
+  const { target, activeCauseSets, activeImpactSets, handleExpandFields } = useContext<
+    IContextState
+  >(FllContext);
+  let fields = tableInfo?.fields;
```

Comment:
don't use optional chaining quite yet. It's still in draft proposal stage and still considered experimental